apps/events.md

```diff
@@ -166,6 +270,40 @@ updatedAt: "2018-04-17T14:48:29.000Z"
 url: "/jam"
 ```
 
+## Search event
+
+To make use of this event, you need to add the following snippet to your code:
+
+```js
+if (window.eComEventTarget) {
+  window.eComEventTarget.addEventListener('search', function (event) {
+    console.log('search', {
+      products: event.detail.products,
+      query: event.detail.query,
+    })
+  })
+}
+```
+
+You'll receive the following information:
+
+```
+product: {
+  size: 5
+  __altered: false
+  __hash: undefined
+  __ownerID: undefined
+  _capacity: 5
+  _level: 5
+  _origin: 0
+  _root: null
+  _tail: Tt {array: Array(5), ownerID: undefined}
+}
```

Comment:
I think this is wrong. It shows ImmutableJS object which is not useful. We should show plain javascript object instead. Am I correct @ingmarh ?

```suggestion
product: {
    availabilityText: "Available"
    available: true
    basePrice: {refQuantity: {…}, refPrice: {…}, formatted: "1 m³ = €0,12", quantity: {…}}
    bulkPrices: null
    conditionMicrodata: "NewCondition"
    customAttributes: [{…}]
    deliveryPeriod: "2-3"
    deliveryPeriodUnit: "DAYS"
    description: null
    energyLabel: null
    energyLabelSourceFile: null
    gtin: 7501054530107
    hasCrossSelling: false
    hasStockLevel: true
    hasVariations: false
    href: "/p/homemade-cherry-jam"
    image: null
    isVariationMaster: false
    isVariationProduct: false
    isVisible: true
    links: (5) [{…}, {…}, {…}, {…}, {…}]
    listPrice: null
    lowestPrice: null
    mainCategoryId: "5954B711-E377-2A90-C400-D5809AB3F62B"
    manufacturer: null
    manufacturerPrice: null
    metaDescription: ""
    name: "Homemade Cherry Jam"
    onStock: true
    outOfStock: false
    pickupPeriod: ""
    price: {taxType: "NET", formatted: "€40,00", amount: 40, currency: "EUR"}
    productDataSheet: null
    productId: "5954B706-E701-F357-A52D-D5809AB3F606"
    productVariationSelection: null
    productVariationValues: ""
    sku: "1007"
    slideshow: [{…}, {…}, {…}, {…}]
    slug: "homemade-cherry-jam"
    stockLevelClass: "in"
    stockLevelMicrodata: "InStock"
    title: "Homemade Cherry Jam"
    variationMaster: null
    variations: null
    vatNote: "components.productComponent.priceExclusiveVat"
    warnStock: false
}
```

Reply:
@bulutfatih I'd also prefer the result you've proposed as it's more easy to understand, but what I included in this PR is what I got when having triggered the event myself... Did you get the another result? Do we need to edit the snippet developers should add to the code in order to get your proposal?

Reply:
I think this also has to do with confusion of different types of objects that one gets depending on which event they are subscribing. Sometimes it is `Immutable.Map`, sometimes a plain JavaScript object. Also just wrote about it in [this other annotation](https://github.com/ePages-de/epages-docs/pull/1190/files#r691347438).

I‘m not sure how to best proceed but I agree with @bulutfatih that if we list the example information here, then it is better to have it shown as plain JavaScript object.

To get a better idea what difference I‘m talking about:

```js
// This is how you access a property on a plain JavaScript object (where `event.detail.product` is the object):
event.detail.product.productId

// This is how you access a property on an Immutable.Map object (where `event.detail.product` is Immutable.Map):
event.detail.product.get('productId')

// And this is how you turn an Immutable.Map object to a plain JavaScript object:
event.detail.product.toJS()

// ... then you can just access "productId" like this:
event.detail.product.toJS().productId
```

I think it is confusing to have both Immutable.Map and plain JavaScript objects on the API that is now documented for app developers.

Reply:
@ingmarh Does it affect the performance much if we turn `detail`_(immutable)_ to plain js object before dispatching? I'm afraid, yes. :/

I agree that we should find a consistence solution before it is late.

@cgebken no, you are right. I also get result with you. I just wanted to mention that result is not useful. Probably we will talk about this in tomorrow's JS Daily.

Reply:
Maybe using `.toJS()` before dispatching is okay for now. However, the API already exists. If there is an app that is already using it, it could be a breaking change. We can talk about it in tomorrow‘s JS Daily.